server/src/test/java/org/elasticsearch/cluster/metadata/ToAndFromJsonMetaDataTests.java

```diff
@@ -246,13 +245,19 @@ public void testSimpleJsonFromAndTo() throws IOException {
         assertThat(indexMetaData.mapping().source().string(), equalTo(MAPPING_SOURCE1));
         assertThat(indexMetaData.getAliases().size(), equalTo(3));
         assertThat(indexMetaData.getAliases().get("alias1").alias(), equalTo("alias1"));
-        assertThat(indexMetaData.getAliases().get("alias1").filter().string(), equalTo(ALIAS_FILTER1));
+        // By default, MetaData.Builder.toXContent(metaData) will use ToXContent.EMPTY_PARAMS, which generates the same serialization
```

Comment:
I think we should remove that static method (MetaData.Builder.toXContent) which seems to be only used by this test.

Instead, we should test both API and non-API serialization in this class, and non-API deserialization.

Reply:
I've moved the new serialization tests from `MetaDataTests` to `ToAndFromJsonMetaDataTests`. 

I've inlined / removed 1 `MetaData.Builder#toXContent` which is only used by test. There's another one though that takes more parameters and used in production code, so I left that one alone.